src/coreclr/jit/gentree.cpp

```diff
@@ -2482,9 +2482,18 @@ bool GenTree::Compare(GenTree* op1, GenTree* op2, bool swapOK)
                         return false;
                     }
                     break;
+
                 case GT_BLK:
                 case GT_OBJ:
-                    if (op1->AsBlk()->GetLayout() != op2->AsBlk()->GetLayout())
+                    if ((op1->gtFlags & (GTF_IND_FLAGS)) != (op2->gtFlags & (GTF_IND_FLAGS)))
+                    {
+                        return false;
+                    }
+                    FALLTHROUGH;
+
+                case GT_IND:
+                case GT_NULLCHECK:
+                    if ((op1->gtFlags & (GTF_IND_FLAGS)) != (op2->gtFlags & (GTF_IND_FLAGS)))
```

Comment:
This code is odd. GT_BLK/GT_OBJ/GT_IND/GT_NULLCHECK are all equivalent; why not share the code? Why does GT_BLK/GT_OBJ have "FALLTHROUGH" to the same code it just executed? also, `(GTF_IND_FLAGS)` doesn't need to be parenthesized.